test/codepage_test.dart

```diff
@@ -62,14 +92,65 @@ void main() {
     expect(decoded, latin2text);
   });
 
-  test('Custom code page', () {
-    var cp = CodePage('custom', "ABCDEF${"\uFFFD" * 250}");
-    var result = cp.encode('BADCAFE');
-    expect(result, [1, 0, 3, 2, 0, 5, 4]);
-    expect(() => cp.encode('GAD'), throwsFormatException);
-    expect(cp.encode('GAD', invalidCharacter: 0x3F), [0x3F, 0, 3]);
-    expect(cp.decode([1, 0, 3, 2, 0, 5, 4]), 'BADCAFE');
-    expect(() => cp.decode([6, 1, 255]), throwsFormatException);
-    expect(cp.decode([6, 1, 255], allowInvalid: true), '\u{FFFD}B\u{FFFD}');
+  group('Custom code page', () {
+    late final CodePage cp;
+
+    setUpAll(() => cp = CodePage('custom', "ABCDEF${"\uFFFD" * 250}"));
```

Comment:
Code pages should be immutable, so just initialize it directly, without using `setUp`. (I also don't like `setUp` :grin:).

Unless you fear that it will throw?

(What's the difference between `setUp` and `setUpAll`?)

Reply:
setUp runs for each embedded test, setUpAll runs once.

Done.